src/processes/linux_list.rs

```diff
@@ -43,6 +61,80 @@ pub fn active_executables() -> Result<ProcessList> {
     Ok(executables.into_values().collect())
 }
 
+pub fn visible_windows() -> Result<HashSet<PID>> {
+    if env::var("WAYLAND_DISPLAY").is_ok() {
+        wayland_visible_windows()
+    } else if env::var("DISPLAY").is_ok() {
+        x11_visible_windows()
+    } else {
+        Ok(HashSet::new())
+    }
+}
+
+fn wayland_visible_windows() -> Result<HashSet<PID>> {
+    // TODO
+    Ok(HashSet::new())
+}
+
+fn x11_visible_windows() -> Result<HashSet<PID>> {
+    let mut pids: HashSet<PID> = HashSet::new();
+
+    let (conn, screen) = x11rb::connect(None)?;
+    // do we want to handle multiple screens as well?
+    let root = conn.setup().roots[screen].root;
+
+    let pid_property_atom = conn.intern_atom(false, b"_NET_WM_PID")?.reply()?.atom;
+    let cardinal_type: Atom = AtomEnum::CARDINAL.into();
+
+    let mut stack: Vec<u32> = Vec::new();
+    stack.push(root);
+
+    while let Some(window) = stack.pop() {
+        // window might not be originating from our machine. see `WM_CLIENT_MACHINE`
+        let tree = conn.query_tree(window)?.reply()?;
+        for child in tree.children {
```

Comment:
If we're handling multiple screens, we should check if the window originates from our local machine (using hostname)